Better error message on missing variable bounds

In JuMP master this code,
```julia
using JuMP
m = Model()
x = @variable(m, [i in 1:3], lower_bound=0)
JuMP.lower_bound(x[1])
JuMP.upper_bound(x[1])
```
leads to the following AssertionError,
```
ERROR: AssertionError: has_upper_bound(v)
Stacktrace:
 [1] upper_bound_index(::JuMP.VariableRef) at /home/.julia/v0.6/JuMP/src/variables.jl:344
 [2] UpperBoundRef at /home/.julia/v0.6/JuMP/src/variables.jl:372 [inlined]
 [3] upper_bound(::JuMP.VariableRef) at /home/.julia/v0.6/JuMP/src/variables.jl:395
```

--- a/src/variables.jl
+++ b/src/variables.jl
@@ -257,7 +257,9 @@ function has_lower_bound(v::VariableRef)
 end
 
 function lower_bound_index(v::VariableRef)
-    @assert has_lower_bound(v) # TODO error message
+    if !has_lower_bound(v)
+        error("Variable $(v) does not have a lower bound.")
+    end
     return owner_model(v).variable_to_lower_bound[index(v)]
 end
 function set_lower_bound_index(v::VariableRef, cindex::MOILB)
@@ -320,7 +322,9 @@ function has_upper_bound(v::VariableRef)
 end
 
 function upper_bound_index(v::VariableRef)
-    @assert has_upper_bound(v) # TODO error message
+    if !has_upper_bound(v)
+        error("Variable $(v) does not have an upper bound.")
+    end
     return owner_model(v).variable_to_upper_bound[index(v)]
 end
 function set_upper_bound_index(v::VariableRef, cindex::MOIUB)

--- a/test/variable.jl
+++ b/test/variable.jl
@@ -205,8 +205,8 @@ function test_variable_bounds_set_get(ModelType)
     @test JuMP.fix_value(fixedvar) == 2.0
     JuMP.fix(fixedvar, 5)
     @test JuMP.fix_value(fixedvar) == 5
-    @test_throws AssertionError JuMP.lower_bound(fixedvar)
-    @test_throws AssertionError JuMP.upper_bound(fixedvar)
+    @test_throws Exception JuMP.lower_bound(fixedvar)
+    @test_throws Exception JuMP.upper_bound(fixedvar)
 end
 
 function test_variable_starts_set_get(ModelType)

